src/reducers/resultReducer.js

```diff
@@ -36,6 +37,10 @@ export default (state = INITIAL_STATE, { type, payload }) => {
       return state.update('results', (results) => results.push(payload));
     case CLEAR_RESULT_SUCCESS:
       return state.setIn(['current', 'content'], Map());
+    case DELETE_RESULT_SUCCESS:
+      return state.updateIn(['results'], (results) =>
+        results.filter((r) => r.id !== payload),
```

Comment:
Even though `results` is right there, I think `results.filter((result) => result.id !== payload)` is clearer